control-plane/agents/core/src/volume/registry.rs

```diff
@@ -35,13 +35,14 @@ impl Registry {
                 size: nexus_state.size,
                 status: match nexus_state.status {
                     NexusStatus::Online
-                        if nexus_state.children.len() != volume_spec.num_replicas as usize =>
+                        if nexus_state.children.len() != volume_spec.num_replicas as usize
+                            || nexus_state.share != volume_spec.protocol =>
```

Comment:
Should the volume status be degraded just because the share protocol is different? The volume status is usually an indication of the health of the data (i.e. the number of replicas) but this would seem to conflate the meaning.

Reply:
I've removed this change.